src/providers.rs

```diff
@@ -217,7 +217,7 @@ pub fn do_unregister_provider(caller: Principal, provider_id: u64) -> bool {
     PROVIDERS.with(|providers| {
         let mut providers = providers.borrow_mut();
         if let Some(provider) = providers.get(&provider_id) {
-            if provider.owner != caller {
+            if !(provider.owner == caller || is_authorized(&caller, Auth::Manage)) {
```

Comment:
Nit: I'm wondering every time I read this: Wouldn't it be easier to read if we removed the negation and switched the `if` and `else` statements?

Reply:
Done.